pkg/render/monitor.go

```diff
@@ -142,6 +171,56 @@ func (mc *monitorComponent) alertmanager() *monitoringv1.Alertmanager {
 	}
 }
 
+func (mc *monitorComponent) alertmanagerSecrets() []*corev1.Secret {
+	// This secret will be mounted as the Alertmanager configuration file.
+	// Source secret is in the tigera-operator namespace and the target secret is in the tigera-prometheus namespace.
+	var secrets []*corev1.Secret
+
+	if mc.alertmanagerConfigSecret != nil {
+		secrets = append(secrets, mc.alertmanagerConfigSecret)
```

Comment:
We should be creating the operator secret in render. If it doesn't exist, it needs to be created in the controller. Only items that are _solely_ managed by the operator should be created here, customers will be interacting directly with this secret. 

This has to do with how we set the owner reference on objects created in the render components.

@tmjd can you remember where we have an example of this? I've been looking for one but I can't find any, I'm not sure if that's a bug or not...

Reply:
I meant to say "shouldn't", I meant the complete opposite of what my comments suggested lol. What you're 2 points are excellent what I meant to get across.